lib/spack/spack/version.py

```diff
@@ -1330,9 +1327,10 @@ def lookup_ref(self, ref):
         known version prior to the commit, as well as the distance from that version
         to the commit in the git repo. Those values are used to compare Version objects.
         """
-        dest = os.path.join(spack.paths.user_repos_cache_path, self.repository_uri)
-        if dest.endswith(".git"):
-            dest = dest[:-4]
+        dest = Path(spack.paths.user_repos_cache_path).joinpath(self.repository_uri)
+
+        dest = dest.with_suffix("")
```

Comment:
Can we use the idiomatic `/` to join paths?

Reply:
Whoops commented this on the wrong line, see line two above.

Reply:
W.R.T. to this line, is removing the file extension _always_ what we want to do here? It looks like the previous behavior was only to conditionally remove the ext when it was `.git`.